rest/rest-sqs/src/main/scala/org/elasticmq/rest/stats/StatisticsDirectives.scala

```diff
@@ -0,0 +1,97 @@
+package org.elasticmq.rest.stats
+
+import java.util.concurrent.Executors
+
+import akka.actor.ActorRef
+import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport
+import akka.http.scaladsl.model.StatusCodes.NotFound
+import akka.http.scaladsl.server.Route
+import org.elasticmq.metrics.QueueMetricsOps
+import org.elasticmq.rest.sqs.QueueAttributesOps
+import org.elasticmq.rest.sqs.directives.ElasticMQDirectives
+import org.elasticmq.util.{NowProvider, NowProviderHolder}
+import org.elasticmq.{QueueData, QueueStatistics}
+import spray.json._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success}
+
+final case class QueuesResponse(
+                                 name: String,
+                                 statistics: QueueStatisticsResponse
+                               )
+
+final case class QueueResponse(
+                                name: String,
+                                attributes: Attributes
+                              )
+
+final case class QueueStatisticsResponse(
+                                          approximateNumberOfVisibleMessages: Long,
+                                          approximateNumberOfInvisibleMessages: Long,
+                                          approximateNumberOfMessagesDelayed: Long
+                                        )
+
+
+trait JsonSupport extends SprayJsonSupport with DefaultJsonProtocol {
+  implicit val queueStatisticsFormat = jsonFormat3(QueueStatisticsResponse)
+  implicit val queuesFormat = jsonFormat2(QueuesResponse)
+  implicit val queueFormat = jsonFormat2(QueueResponse)
+}
+
+trait StatisticsDirectives extends JsonSupport {
+  this: ElasticMQDirectives with QueueAttributesOps with NowProviderHolder =>
+
+  lazy val ec: ExecutionContext = ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(2))
+  implicit val duration = timeout.duration
+
+  val np: NowProvider = nowProvider
```

Comment:
Preference opinion:
For me, it is clearer to pass dependencies explicitly instead of depending on the inheritance.
For example, in `StatisticsRestServerBuilder` fact that Statistics endpoints require dependency on time abstraction is hidden. On the other hand, if `statistics` method would require explicitly `NowProvider` then it is visible at first glance that there is such dependency in this functionality.

Reply:
Yeah, this will be much simpler. Thanks